include/deal.II/base/array_view.h

```diff
@@ -286,6 +349,52 @@ make_array_view (std::vector<ElementType> &vector,
 }
 
 
+/**
+ * Create an ArrayView that takes a pair of iterators as arguments. The type
+ * of the ArrayView is inferred from the value type of the iterator (e.g., the
+ * view created from two const iterators will have a const type).
+ *
+ * @warning This function only makes sense for array or vector iterators
+ * (i.e., the memory between @p begin and @end contains valid objects). This
+ * function is intended for use with the various boost vector classes and
+ * <code>std::vector</code>, but not, e.g., <code>std::deque</code>.
+ *
+ * @relates ArrayView
+ */
+template <typename Iterator>
+ArrayView<typename std::remove_reference<typename std::iterator_traits<Iterator>::reference>::type>
+make_array_view (const Iterator begin, const Iterator end)
```

Comment:
I think it would be better not to support this case since, to the best of my knowledge, if `begin()` and `end()` return different types then the container is not some glorified array wrapper and creating an array view doesn't make sense.